dds_web/web/user.py

```diff
@@ -320,8 +320,10 @@ def confirm_2fa():
         # Raises authenticationerror if invalid
         try:
             twofactor_verify(twofactor_value.encode())
-        except ddserr.AuthenticationError:
-            flask.flash("Invalid one-time code.", "warning")
+        except ddserr.AuthenticationError as err:
+            message = str(err)
+            message = message.removeprefix("401 Unauthorized: ")
+            flask.flash(message, "warning")
```

Comment:
I think we need to check what the message is here first. `twofactor_verify` (in this case `verify_TOTP`) raises an AuthenticationError both if it's invalid and if it's within 90 seconds. So maybe check if the string contains "90 seconds" or something? And if it does --> do what you've done here, and if it doesn't then print the original message ("Invalid one-time code")

Reply:
Oh, then never mind!

Reply:
Oh, then never mind!